### ISSUE TITLE
Regression Caused by Removing Parantheses

### ISSUE THREAD

**Issue by peter-empen** [scalameta/scalafmt#3228]
## Configuration (required) ##

Please paste the contents of your `.scalafmt.conf` file here:
```
version=3.5.3
runner.dialect = scala213

align.preset = more
align.openParenCallSite = false
assumeStandardLibraryStripMargin = true
importSelectors = binPack
indentOperator.exclude = "^(and|~|&&|\\|\\||\\+|\\-)$"
maxColumn = 120
newlines {
  alwaysBeforeMultilineDef = false
  neverInResultType = true
  penalizeSingleSelectMultiArgList = false
}
optIn.breakChainOnFirstMethodDot = false
rewrite.rules = [PreferCurlyFors, RedundantBraces, RedundantParens, SortModifiers, SortImports]

docstrings.style = SpaceAsterisk
docstrings.wrap = no

project.excludeFilters = [".*/scala/collection/mutable/ExtHashSet.scala"]
```

## Command-line parameters (required) ##

When I run scalafmt via CLI like this: `scalafmt ./core/src/main/scala/scalax/collection/mutable/Graph.scala`
See also [Update scalafmt-core to 3.5.3](https://github.com/scala-graph/scala-graph/pull/234/files).

## Steps

Given code like this:
```scala
// constructor has two parameter lists with the first containing default values for all parameters
new DefaultGraphImpl[N, E]()(config)
```

## Problem

Scalafmt formats code like this:
```scala
// no more compiling
new DefaultGraphImpl[N, E](config)
```

## Expectation

I would like the formatted output to look like this:
```scala
// don't remove empty parameter list in this case
new DefaultGraphImpl[N, E]()(config)
```

## Workaround

```
// I need to supply at least one parameter in the first parameter list
new DefaultGraphImpl[N, E](Set.empty)(config)
```

## Notes

For a similar case, see also core/src/main/scala/scalax/collection/mutable/EqHash.scala line 154 in [Update scalafmt-core to 3.5.3](https://github.com/scala-graph/scala-graph/pull/234/files).

**Comment by kitbellew** [comment]
@peter-empen 3.5.3 is not the latest version. please reopen if applicable in 3.5.4.

**Comment by peter-empen** [comment]
I did test it via CLI 3.5.4 as requested. Not sure how to reopen, it seems I'm not allowed to.

**Comment by kitbellew** [comment]
please confirm that you changed the `version` string in your config file. a very similar problem was fixed in 3.5.4.

**Comment by peter-empen** [comment]
After upgrading to 3.5.4 the major issue has vanished :+1:
